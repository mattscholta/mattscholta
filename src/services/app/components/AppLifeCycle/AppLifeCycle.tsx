import * as React from 'react';

import { setTrackingDefaults } from '@src/utils/tracking';

interface AppLifeCycleProps {}

/**
 * @name AppLifeCycle
 * @description We use this component to mount any Application specific
 * logic that we need to initialize 1 time only while still having access to
 * both Apollo and Recoil state.
 *
 * We're using it for mount and un-mount
 */
const AppLifeCycle: React.FC<AppLifeCycleProps> = (_props) => {
  // Life Cycle
  React.useEffect(() => {
    setTrackingDefaults();
  }, []);

  return null;
};

export { AppLifeCycle };
export type { AppLifeCycleProps };
