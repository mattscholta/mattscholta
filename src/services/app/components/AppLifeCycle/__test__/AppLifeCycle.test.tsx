import * as React from 'react';
import { render, RenderResult } from '@testing-library/react';
import { RecoilRoot } from 'recoil';

import { AppLifeCycle, AppLifeCycleProps } from '../AppLifeCycle';

describe('AppLifeCycle', () => {
  const gtag = jest.fn();

  let props: AppLifeCycleProps;
  let wrapper: RenderResult;

  const createWrapper = () => (
    <RecoilRoot>
      <AppLifeCycle {...props} />
    </RecoilRoot>
  );

  beforeAll(() => {
    window.gtag = gtag;
  });

  beforeEach(() => {
    props = {};
    wrapper = render(createWrapper());
  });

  it('matches snapshot', () => {
    expect(wrapper.container.childNodes).toMatchSnapshot();
  });
});
