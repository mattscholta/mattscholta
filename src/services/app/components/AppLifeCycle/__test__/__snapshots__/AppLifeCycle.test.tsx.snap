// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLifeCycle matches snapshot 1`] = `NodeList []`;
